ReactiveCocoaTests/Swift/SignalSpec.swift

```diff
@@ -65,57 +65,282 @@ class SignalSpec: QuickSpec {
+				let (baseSignal, sink) = Signal<Int, NoError>.pipe()
+				let signal = baseSignal |> skip(0)
+
+				var lastValue: Int?
+				signal.observe(next: { lastValue = $0 })
+
+				expect(lastValue).to(beNil())
+
+				sendNext(sink, 1)
+				expect(lastValue).to(equal(1))
+
+				sendNext(sink, 2)
+				expect(lastValue).to(equal(2))
 			}
 		}
 
 		describe("skipRepeats") {
-			pending("should skip duplicate Equatable values") {
+			it("should skip duplicate Equatable values") {
+				let (baseSignal, sink) = Signal<Bool, NoError>.pipe()
+				let signal = baseSignal |> skipRepeats
+
+				var values: [Bool] = []
+				signal.observe(next: { values.append($0) })
+
+				expect(values).to(equal([]))
+
+				sendNext(sink, true)
+				expect(values).to(equal([true]))
+
+				sendNext(sink, true)
+				expect(values).to(equal([true]))
+
+				sendNext(sink, false)
+				expect(values).to(equal([true, false]))
+
+				sendNext(sink, true)
+				expect(values).to(equal([true, false, true]))
 			}
 
-			pending("should skip values according to a predicate") {
+			it("should skip values according to a predicate") {
+				let (baseSignal, sink) = Signal<String, NoError>.pipe()
+				let signal = baseSignal |> skipRepeats { countElements($0) == countElements($1) }
+
+				var values: [String] = []
+				signal.observe(next: { values.append($0) })
+
+				expect(values).to(equal([]))
+
+				sendNext(sink, "a")
+				expect(values).to(equal(["a"]))
+
+				sendNext(sink, "b")
+				expect(values).to(equal(["a"]))
+
+				sendNext(sink, "cc")
+				expect(values).to(equal(["a", "cc"]))
+
+				sendNext(sink, "d")
+				expect(values).to(equal(["a", "cc", "d"]))
 			}
 		}
 
 		describe("skipWhile") {
-			pending("should skip while the predicate is true") {
+			var signal: Signal<Int, NoError>!
+			var sink: Signal<Int, NoError>.Observer!
+
+			var lastValue: Int?
+
+			beforeEach {
+				let (baseSignal, observer) = Signal<Int, NoError>.pipe()
+
+				signal = baseSignal |> skipWhile { $0 < 2 }
+				sink = observer
+				lastValue = nil
+
+				signal.observe(next: { lastValue = $0 })
 			}
 
-			pending("should not skip any values when the predicate starts false") {
+			it("should skip while the predicate is true") {
+				expect(lastValue).to(beNil())
+
+				sendNext(sink, 1)
+				expect(lastValue).to(beNil())
+
+				sendNext(sink, 2)
+				expect(lastValue).to(equal(2))
+
+				sendNext(sink, 0)
+				expect(lastValue).to(equal(0))
+			}
+
+			it("should not skip any values when the predicate starts false") {
+				expect(lastValue).to(beNil())
+
+				sendNext(sink, 3)
+				expect(lastValue).to(equal(3))
+
+				sendNext(sink, 1)
+				expect(lastValue).to(equal(1))
 			}
 		}
 
 		describe("take") {
-			pending("should take initial values") {
+			it("should take initial values") {
+				let (baseSignal, sink) = Signal<Int, NoError>.pipe()
+				let signal = baseSignal |> take(2)
+
+				var lastValue: Int?
+				var completed = false
+				signal.observe(
+					next: { lastValue = $0 },
```

Comment:
Can you please remain consistent with the style for `observe` that we've used elsewhere?